File: truckdoc-client-m2/application/src/main/java/com/sanda/truckdoc/client/data/model/DbContactRecord.java
package com.sanda.truckdoc.client.data.model;

import android.graphics.Color;

import com.github.naixx.WithId;
import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;
import com.sanda.truckdoc.client.api.model.ContactListAttribute;
import com.sanda.truckdoc.client.api.model.ContactRecord;

@DatabaseTable(tableName = "contact_records")
public class DbContactRecord implements WithId {

    @DatabaseField(id = true)
    private Long recipientId;
    @DatabaseField
    private String label;
    @DatabaseField
    private String recipientIdType;
    @DatabaseField
    private int color;
    @DatabaseField
    private String phone;

    public DbContactRecord() {
    }

    public DbContactRecord(Long recipientId, String label, String recipientIdType, int color, String phone) {
        this.recipientId = recipientId;
        this.label = label;
        this.recipientIdType = recipientIdType;
        this.color = color;
        this.phone = phone;
    }

    public DbContactRecord(ContactRecord contactRecord) {
        recipientId = contactRecord.getRecipientId();
        label = contactRecord.getLabel();
        recipientIdType = contactRecord.getRecipientIdType();
        if (contactRecord.getAttributes() != null && contactRecord.getAttributes().containsKey(ContactListAttribute.COLOR.getKey())) {
            color = Color.parseColor(contactRecord.getAttributes().get(ContactListAttribute.COLOR.getKey()));
        } else {
            color = Color.WHITE;
        }
    }

    public Long getRecipientId() {
        return recipientId;
    }

    @Override
    public long getId() {
        return recipientId;
    }

    public String getLabel() {
        return label;
    }

    public String getRecipientIdType() {
        return recipientIdType;
    }

    public int getColor() {
        return color;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }
}

File: truckdoc-client-m2/application/src/main/java/com/sanda/truckdoc/client/data/model/DbContactRecord.kt
package com.sanda.truckdoc.client.data.model

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "contact_records",
    indices = [Index(value = ["phone"], unique = true)]
)
data class DbContactRecord(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val label: String = "",
    val phone: String = "",
    val role: String = "",
    val recipientId: Long = 0L,
    val recipientIdType: String = ""
) {
    // Constructor for Java compatibility
    constructor(id: Int, label: String, phone: String, role: String) : this(
        id = id,
        label = label,
        phone = phone,
        role = role,
        recipientId = 0L,
        recipientIdType = ""
    )
    
    fun getColor(): Int = 0xFF0000.toInt() // Default red color
    fun getRecordId(): Long = id.toLong()
    
    // Java compatibility methods
    fun setPhone(phone: String): DbContactRecord = copy(phone = phone)
    fun setLabel(label: String): DbContactRecord = copy(label = label)
    fun setRole(role: String): DbContactRecord = copy(role = role)
} 